tests/drivers/uart/uart_async_api/src/test_uart_async.c

```diff
@@ -74,13 +74,13 @@ void test_single_read(void)
 			  "Initial buffer check failed");
 
 	uart_rx_enable(uart_dev, rx_buf, 10, 50);
-	zassert_equal(k_sem_take(&rx_rdy, K_MSEC(100)), -EAGAIN,
+	zassert_equal(k_sem_take(&rx_rdy, K_MSEC(10)), -EAGAIN,
```

Comment:
At least on nrf drivers RX timeout was implemented as break between bytes. It allows to report user when some data was received. It seems like more reasonable than timeouting even without any bytes received. In that case 100ms is an arbitrary value.

However, i see that api is not clear about it.